File: backend/src/main/kotlin/ch/streckeisen/mycv/backend/dashboard/DashboardInfoDto.kt
package ch.streckeisen.mycv.backend.dashboard

import com.fasterxml.jackson.annotation.JsonProperty

data class DashboardInfoDto(
    @get:JsonProperty(value = "isVerified")
    val isVerified: Boolean,
    val profile: ProfileInfoDto?,
    val applications: List<ApplicationInfoDto>
)

File: frontend/src/types/dashboard/DashboardInfoDto.ts
import { ApplicationStatusDto } from '@/types/application/ApplicationStatusDto.ts'

export interface DashboardInfoDto {
  isVerified: boolean
  profile: ProfileInfoDto
  applications: ApplicationInfoDto[]
}

export interface ProfileInfoDto {
  experienceCount: number
  educationCount: number
  projectCount: number
  skillCount: number
}

export interface ApplicationInfoDto {
  status: ApplicationStatusDto
  count: number
}